reconstruct/reconstruct.go

```diff
@@ -101,9 +105,95 @@ func ReconAccuracy(simFilename string, reconFilename string, leavesOnlyFile stri
 	acc := 100 - accuracy
 	answer[allNodes] = acc
 
+	return answer, accByBase
+}
+
+// ReconAccuracyByBase will run if the option given to ReconAccuracy = 1. This function calculates the percentage of first, second and third bases in codons that were correct.
+func ReconAccuracyByBase(simFilename string, reconFilename string, gpFilename string) map[string][]float64 {
+	sim := fasta.Read(simFilename)
+	recon := fasta.Read(reconFilename)
+	genes := genePred.Read(gpFilename)
+	answer := make(map[string][]float64)
+
+	for s := 0; s < len(sim); s++ {
+		for r := 0; r < len(recon); r++ {
+			var percentage1 float64
+			var percentage2 float64
+			var percentage3 float64
```

Comment:
Consider merging these variable declarations to the same line.
e.g. `var percentage1, percentage2, percentage3 float64`